windows/src/engine/keyman32/appint/aiTIP.cpp

```diff
@@ -100,14 +100,17 @@ void ProcessToggleChange(UINT key) {   // I4793
   switch(key) {  
     case VK_CAPITAL: flag = CAPITALFLAG; break;
     case VK_NUMLOCK: flag = NUMLOCKFLAG; break;
+    default: return;
   }
 
-  if(*Globals::ShiftState() & flag)  {
-    *Globals::ShiftState() &= ~flag; 
-  }
-  else { 
+  if (GetKeyState(key) & 1) {
```

Comment:
This reads the current key state rather than relying on a local flag. It's a bad idea to maintain two flags any more than you have to :)